src/network/vector_write.cpp

```diff
@@ -0,0 +1,45 @@
+#include "network/vector_write.h"
+
+#include <cassert>
+#include <cerrno>
+#include <cstring>
+#include <sstream>
+#include <stdexcept>
+
+#include <alloca.h>
+#include <sys/uio.h>
+
+namespace lightstep {
+//--------------------------------------------------------------------------------------------------
+// Write
+//--------------------------------------------------------------------------------------------------
+std::tuple<int, int, int> Write(
+    int socket, std::initializer_list<const FragmentSet*> fragment_sets) {
+  int num_fragments = 0;
```

Comment:
Is there any concern about exceeding the limit of `num_fragments`? I believe the limit is 1024 on linux systems. I doubt this is a real concern, but wonder what your take is.

Reply:
It shouldn't be an issue currently since most of the spans will be stored contiguously.

It might come up when we switch to a higher performance arena allocation approach, but should be easily solvable with a loop.

I'll add an assert statement for now.